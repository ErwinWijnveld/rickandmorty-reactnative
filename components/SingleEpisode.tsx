import { StyleSheet, Text, View } from 'react-native';
import { EpisodeProps } from '../@types/episodes';
import { COLORS } from '../lib/consts';
const SingleEpisode = ({ name, air_date, episode }: EpisodeProps) => {
	return (
		<View style={styles.container}>
			<View>
				{name && <Text>{name}</Text>}
				{air_date && <Text>{air_date}</Text>}
				{episode && <Text>{episode}</Text>}
			</View>
		</View>
	);
};
export default SingleEpisode;
const styles = StyleSheet.create({
	container: {
		backgroundColor: COLORS.greyTint,
		borderRadius: 10,
		padding: 10,
		marginBottom: 10,
		display: 'flex',
		flexDirection: 'row',
		alignItems: 'center',
	},
	count: {
		fontSize: 20,
		fontWeight: 'bold',
		marginRight: 20,
		marginLeft: 10,
		flexShrink: 0,
	},
});
