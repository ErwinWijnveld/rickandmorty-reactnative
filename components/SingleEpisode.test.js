import { create } from "react-test-renderer";
import SingleEpisode from "./SingleEpisode";

describe("SingleEpisode", () => {
    it("renders correctly with missing props", () => {
        const props = {
            name: "Test",
            air_date: "2020-01-01",
        }
        const tree = create(
            <SingleEpisode
                {...props}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

