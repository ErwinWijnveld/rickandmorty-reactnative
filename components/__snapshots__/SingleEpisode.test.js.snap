// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleEpisode renders correctly with missing props 1`] = `
<View>
  <Text>
    Test
  </Text>
  <Text>
    2020-01-01
  </Text>
</View>
`;
